core/pkg/launch/job_builder.go

```diff
@@ -711,6 +712,9 @@ func (j *JobBuilder) makeJobMetadata(output *data_types.TypeRepresentation) (str
 			j.logger.Debug("jobBuilder: error inferring run config types", err)
 		}
 	}
+	for _, configFile := range j.configFiles {
+		metadata[configFile.relpath] = j.generateConfigFileSchema(configFile)
+	}
 	metadata = map[string]interface{}{"input_types": metadata}
```

Comment:
what's going on here? isn't this overwriting all the `metadata` updates on the previous lines?

Reply:
We are only setting new keys. But I have changed this anyway so that we store the file schema under `input_types.files.<file-relpath>` instead of just `input_types` and I think the code got much clearer